### ISSUE TITLE
No environment name when I use --prefix

### ISSUE THREAD

**Issue by thatlittleboy** [conda/conda#7831]
**I'm submitting a...**
  - [ ] bug report
  - [ ] feature request

I'm not quite sure, it's possibly just a simple question.

## Current Behavior

I want to create a conda environment in a different folder (say, `D:\user\Documents\fastai`) from the default `.conda`.

After a `conda create --prefix` and `conda env update`, the environment gets created correctly but without an "alias"/"name" so activating it is a hassle. Now I have to do `conda activate D:\\user\\Documents\\fastai\\fastai_env` or similar. 

Is there a command / prefix / argument I'm missing?

![chrome_2018-10-06_01-55-40](https://user-images.githubusercontent.com/30731072/46567942-f3169700-c90a-11e8-99ab-0f7dcfd30a00.png)

### Steps to Reproduce

While I'm in the folder I want to create the environment in, I execute these two in succession:
```
conda create --prefix=fastai_env python=3.6 anaconda
conda env update --prefix=fastai_env
```
Then `conda env list` gives the above result in the picture.

## Expected Behavior

I want to be able to specify a name for the environment just like what would happen if I were to use `conda create -n <env name> ...` for ease of activation.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : C:\ProgramData\Anaconda3
            shell level : 1
       user config file : C:\Users\user\.condarc
 populated config files : C:\Users\user\.condarc
          conda version : 4.5.11
    conda-build version : 3.15.1
         python version : 3.6.6.final.0
       base environment : C:\ProgramData\Anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/win-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\ProgramData\Anaconda3\pkgs
                          C:\Users\user\AppData\Local\conda\conda\pkgs
       envs directories : C:\ProgramData\Anaconda3\envs
                          C:\Users\user\AppData\Local\conda\conda\envs
                          C:\Users\user\.conda\envs
               platform : win-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.6.6 Windows/10 Windows/10.0.17134
          administrator : True
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> C:\Users\user\.condarc <==
ssl_verify: True
channels:
  - defaults
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\ProgramData\Anaconda3:
#
# Name                    Version                   Build  Channel
_ipyw_jlab_nb_ext_conf    0.1.0            py36he6757f0_0    defaults
_tflow_1100_select        0.0.3                       mkl    defaults
_tflow_190_select         0.0.2                     eigen    defaults
absl-py                   0.4.1                    py36_0    defaults
alabaster                 0.7.11                   py36_0    defaults
anaconda                  custom           py36h363777c_0    defaults
anaconda-client           1.7.2                    py36_0    defaults
anaconda-navigator        1.8.7                    py36_0    defaults
anaconda-project          0.8.2                    py36_0    defaults
appdirs                   1.4.3            py36h28b3542_0    defaults
asn1crypto                0.24.0                   py36_0    defaults
astor                     0.7.1                    py36_0    defaults
astroid                   2.0.4                    py36_0    defaults
attrs                     18.2.0           py36h28b3542_0    defaults
autobahn                  18.3.1                     py_0    vpython
automat                   0.7.0                    py36_0    defaults
babel                     2.6.0                    py36_0    defaults
backcall                  0.1.0                    py36_0    defaults
backports                 1.0                      py36_1    defaults
backports.shutil_get_terminal_size 1.0.0                    py36_2    defaults
beautifulsoup4            4.6.3                    py36_0    defaults
bitarray                  0.8.3            py36hfa6e2cd_0    defaults
bkcharts                  0.2              py36h7e685f7_0    defaults
blas                      1.0                         mkl    defaults
blaze                     0.11.3           py36h8a29ca5_0    defaults
bleach                    2.1.4                    py36_0    defaults
blosc                     1.14.4               he51fdeb_0    defaults
boto                      2.49.0                   py36_0    defaults
bottleneck                1.2.1            py36h452e1ab_1    defaults
branca                    0.2.0                     <pip>
bzip2                     1.0.6                hfa6e2cd_5    defaults
ca-certificates           2018.8.24            ha4d7672_0    conda-forge
cachecontrol              0.12.5                   py36_0    defaults
certifi                   2018.8.24                py36_0    conda-forge
cffi                      1.11.5           py36h74b6da3_1    defaults
chardet                   3.0.4                    py36_1    defaults
click                     6.7              py36hec8c647_0    defaults
cloudpickle               0.5.5                    py36_0    defaults
clyent                    1.2.2                    py36_1    defaults
colorama                  0.3.9            py36h029ae33_0    defaults
colorlog                  3.1.4                     <pip>
comtypes                  1.1.7                    py36_0    defaults
conda                     4.5.11                   py36_0    conda-forge
conda-build               3.15.1                   py36_0    defaults
conda-env                 2.6.0                         1    defaults
conda-verify              3.1.0                    py36_0    defaults
console_shortcut          0.1.1                         3    defaults
constantly                15.1.0           py36h28b3542_0    defaults
contextlib2               0.5.5            py36he5d52c0_0    defaults
cryptography              2.3.1            py36h74b6da3_0    defaults
curl                      7.61.0               h7602738_0    defaults
cycler                    0.10.0           py36h009560c_0    defaults
cython                    0.28.5           py36h6538335_0    defaults
cytoolz                   0.9.0.1          py36hfa6e2cd_1    defaults
dask-core                 0.19.1                   py36_0    defaults
datashape                 0.5.4                    py36_1    defaults
decorator                 4.3.0                     <pip>
decorator                 4.3.0                    py36_0    defaults
dill                      0.2.8.2                  py36_0    defaults
distlib                   0.2.7                    py36_0    defaults
docutils                  0.14             py36h6012d8f_0    defaults
eli5                      0.8                        py_0    conda-forge
entrypoints               0.2.3                    py36_2    defaults
et_xmlfile                1.0.1            py36h3d2d736_0    defaults
fastcache                 1.0.2            py36hfa6e2cd_2    defaults
filelock                  3.0.8                    py36_0    defaults
flask                     1.0.2                    py36_1    defaults
flask-cors                3.0.6                    py36_0    defaults
folium                    0.5.0                     <pip>
freetype                  2.9.1                ha9979f8_1    defaults
future                    0.16.0                   py36_2    defaults
gast                      0.2.0                    py36_0    defaults
get_terminal_size         1.0.0                h38e98db_0    defaults
gevent                    1.3.6            py36hfa6e2cd_0    defaults
gitdb2                    2.0.3                     <pip>
GitPython                 2.1.8                     <pip>
glob2                     0.6                      py36_0    defaults
graphviz                  2.38.0               h6538335_9    conda-forge
greenlet                  0.4.15           py36hfa6e2cd_0    defaults
grpcio                    1.12.1           py36h1a1b453_0    defaults
h5py                      2.8.0            py36h3bdd7fb_2    defaults
hdf5                      1.10.2               hac2f561_1    defaults
heapdict                  1.0.0                    py36_2    defaults
html5lib                  1.0.1                    py36_0    defaults
hyperlink                 18.0.0                   py36_0    defaults
icc_rt                    2017.0.4             h97af966_0    defaults
icu                       58.2                 ha66f8fd_1    defaults
idna                      2.7                      py36_0    defaults
imageio                   2.4.1                    py36_0    defaults
imagesize                 1.1.0                    py36_0    defaults
incremental               17.5.0                   py36_0    defaults
intel-openmp              2018.0.3                      0    defaults
ipykernel                 4.9.0                    py36_1    defaults
ipython                   6.5.0                    py36_0    defaults
ipython_genutils          0.2.0            py36h3c5d0ee_0    defaults
ipywidgets                7.4.1                    py36_0    defaults
itsdangerous              0.24                     py36_1    defaults
jdcal                     1.4                      py36_0    defaults
jedi                      0.12.1                   py36_0    defaults
jinja2                    2.10                     py36_0    defaults
joblib                    0.12.3                   py36_0    defaults
jpeg                      9b                   hb83a4c4_2    defaults
jsonschema                2.6.0            py36h7636477_0    defaults
jupyter                   1.0.0                    py36_7    defaults
jupyter_client            5.2.3                    py36_0    defaults
jupyter_console           5.2.0                    py36_1    defaults
jupyter_core              4.4.0                    py36_0    defaults
keras                     2.2.2                         0    defaults
keras-applications        1.0.4                    py36_1    defaults
keras-base                2.2.2                    py36_0    defaults
keras-preprocessing       1.0.2                    py36_1    defaults
kiwisolver                1.0.1            py36h6538335_0    defaults
lazy-object-proxy         1.3.1            py36hfa6e2cd_2    defaults
libcurl                   7.61.0               h7602738_0    defaults
libgpuarray               0.7.6                hfa6e2cd_0    defaults
libiconv                  1.15                 h1df5818_7    defaults
libmklml                  2018.0.3                      1    defaults
libpng                    1.6.34               h79bbb47_0    defaults
libprotobuf               3.6.0                h1a1b453_0    defaults
libpython                 2.1                      py36_0    defaults
libsodium                 1.0.16               h9d3ae62_0    defaults
libssh2                   1.8.0                hd619d38_4    defaults
libtiff                   4.0.9                h36446d0_2    defaults
libxml2                   2.9.8                hadb2253_1    defaults
libxslt                   1.1.32               hf6f1972_0    defaults
llvmlite                  0.24.0           py36h6538335_0    defaults
locket                    0.2.0            py36hfed976d_1    defaults
lockfile                  0.12.2           py36h0468280_0    defaults
lxml                      4.2.5            py36hef2cd61_0    defaults
lzo                       2.10                 h6df0209_2    defaults
m2w64-binutils            2.25.1                        5    defaults
m2w64-bzip2               1.0.6                         6    defaults
m2w64-crt-git             5.0.0.4636.2595836               2    defaults
m2w64-gcc                 5.3.0                         6    defaults
m2w64-gcc-ada             5.3.0                         6    defaults
m2w64-gcc-fortran         5.3.0                         6    defaults
m2w64-gcc-libgfortran     5.3.0                         6    defaults
m2w64-gcc-libs            5.3.0                         7    defaults
m2w64-gcc-libs-core       5.3.0                         7    defaults
m2w64-gcc-objc            5.3.0                         6    defaults
m2w64-gmp                 6.1.0                         2    defaults
m2w64-headers-git         5.0.0.4636.c0ad18a               2    defaults
m2w64-isl                 0.16.1                        2    defaults
m2w64-libiconv            1.14                          6    defaults
m2w64-libmangle-git       5.0.0.4509.2e5a9a2               2    defaults
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    defaults
m2w64-make                4.1.2351.a80a8b8               2    defaults
m2w64-mpc                 1.0.3                         3    defaults
m2w64-mpfr                3.1.4                         4    defaults
m2w64-pkg-config          0.29.1                        2    defaults
m2w64-toolchain           5.3.0                         7    defaults
m2w64-tools-git           5.0.0.4592.90b8472               2    defaults
m2w64-windows-default-manifest 6.4                           3    defaults
m2w64-winpthreads-git     5.0.0.4634.697f757               2    defaults
m2w64-zlib                1.2.8                        10    defaults
mako                      1.0.7                    py36_0    defaults
markdown                  2.6.11                   py36_0    defaults
markupsafe                1.0              py36hfa6e2cd_1    defaults
matplotlib                2.2.3            py36hd159220_0    defaults
mccabe                    0.6.1                    py36_1    defaults
menuinst                  1.4.14           py36hfa6e2cd_0    defaults
mistune                   0.8.3            py36hfa6e2cd_1    defaults
mkl                       2019.0                      118    defaults
mkl-service               1.1.2            py36hb217b18_5    defaults
mkl_fft                   1.0.4            py36h1e22a9b_1    defaults
mkl_random                1.0.1            py36h77b88f5_1    defaults
more-itertools            4.3.0                    py36_0    defaults
mpmath                    1.0.0                    py36_2    defaults
msgpack-python            0.5.6            py36he980bc4_1    defaults
msys2-conda-epoch         20160418                      1    defaults
multipledispatch          0.6.0                    py36_0    defaults
navigator-updater         0.2.1                    py36_0    defaults
nbconvert                 5.3.1                    py36_0    defaults
nbdime                    0.4.1                     <pip>
nbformat                  4.4.0            py36h3a5bc1b_0    defaults
networkx                  2.1                      py36_0    defaults
nltk                      3.3.0                    py36_0    defaults
nose                      1.3.7                    py36_2    defaults
notebook                  5.6.0                    py36_0    defaults
numba                     0.39.0           py36h830ac7b_0    defaults
numexpr                   2.6.8            py36h9ef55f4_0    defaults
numpy                     1.15.1           py36ha559c80_0    defaults
numpy-base                1.15.1           py36h8128ebf_0    defaults
numpydoc                  0.8.0                    py36_0    defaults
odo                       0.5.1            py36h7560279_0    defaults
olefile                   0.46                     py36_0    defaults
openssl                   1.0.2p            hfa6e2cd_1000    conda-forge
packaging                 17.1                     py36_0    defaults
pandas                    0.23.4           py36h830ac7b_0    defaults
pandoc                    2.2.3.2                       0    defaults
pandocfilters             1.4.2                    py36_1    defaults
parso                     0.3.1                    py36_0    defaults
partd                     0.3.8            py36hc8e763b_0    defaults
path.py                   11.1.0                   py36_0    defaults
pathlib2                  2.3.2                    py36_0    defaults
patsy                     0.5.0                    py36_0    defaults
PDPbox                    0.2.0                     <pip>
pep8                      1.7.1                    py36_0    defaults
pickleshare               0.7.4            py36h9de030f_0    defaults
pillow                    5.2.0            py36h08bbbbd_0    defaults
pip                       10.0.1                   py36_0    defaults
pip                       10.0.1                    <pip>
pip                       18.0                      <pip>
pkginfo                   1.4.2                    py36_1    defaults
pluggy                    0.7.1            py36h28b3542_0    defaults
ply                       3.11                     py36_0    defaults
progress                  1.4                      py36_0    defaults
prometheus_client         0.3.1            py36h28b3542_0    defaults
prompt_toolkit            1.0.15           py36h60b8f86_0    defaults
protobuf                  3.6.0            py36he025d50_0    defaults
psutil                    5.4.7            py36hfa6e2cd_0    defaults
py                        1.6.0                    py36_0    defaults
pyasn1                    0.4.4            py36h28b3542_0    defaults
pyasn1-modules            0.2.2                    py36_0    defaults
pycodestyle               2.4.0                    py36_0    defaults
pycosat                   0.6.3            py36hfa6e2cd_0    defaults
pycparser                 2.18                     py36_1    defaults
pycrypto                  2.6.1            py36hfa6e2cd_9    defaults
pycurl                    7.43.0.2         py36h74b6da3_0    defaults
pyflakes                  2.0.0                    py36_0    defaults
pygments                  2.2.0            py36hb010967_0    defaults
pygpu                     0.7.6            py36h452e1ab_0    defaults
pylint                    1.8.2                    py36_0    defaults
pyodbc                    4.0.24           py36h6538335_0    defaults
pyopenssl                 18.0.0                   py36_0    defaults
pyparsing                 2.2.0                    py36_1    defaults
pyqt                      5.9.2            py36ha878b3d_0    defaults
pyreadline                2.1                      py36_1    defaults
pysocks                   1.6.8                    py36_0    defaults
pytables                  3.4.4            py36he6f6034_0    defaults
pytest-arraydiff          0.2                      py36_0    defaults
pytest-astropy            0.2.1                    py36_0    defaults
pytest-doctestplus        0.1.2                    py36_0    defaults
pytest-openfiles          0.2.0                    py36_0    defaults
pytest-remotedata         0.2.0                    py36_0    defaults
python                    3.6.6                hea74fb7_0    defaults
python-dateutil           2.7.3                    py36_0    defaults
python-graphviz           0.8.4                    py36_2    conda-forge
pytz                      2018.5                   py36_0    defaults
pywavelets                1.0.0            py36h452e1ab_0    defaults
pywin32                   223              py36hfa6e2cd_1    defaults
pywinpty                  0.5.4                    py36_0    defaults
pyyaml                    3.13             py36hfa6e2cd_0    defaults
pyzmq                     17.1.2           py36hfa6e2cd_0    defaults
qt                        5.9.6            vc14h1e9a669_2    defaults
qtawesome                 0.4.4            py36h5aa48f6_0    defaults
qtconsole                 4.4.1                    py36_0    defaults
qtpy                      1.5.0                    py36_0    defaults
quspin                    0.3.0            py36ha6300db_0    weinbe58
requests                  2.19.1                   py36_0    defaults
rise                      5.1.0                    py36_1    damianavila82
rope                      0.11.0                   py36_0    defaults
ruamel_yaml               0.15.46          py36hfa6e2cd_0    defaults
scikit-image              0.13.1           py36hfa6e2cd_1    defaults
scikit-learn              0.19.2           py36heebcf9a_0    defaults
scipy                     1.1.0            py36h4f6bf74_1    defaults
seaborn                   0.9.0                    py36_0    defaults
send2trash                1.5.0                    py36_0    defaults
service_identity          17.0.0           py36h28b3542_0    defaults
setuptools                40.2.0                   py36_0    defaults
shap                      0.24.0                    <pip>
simplegeneric             0.8.1                    py36_2    defaults
singledispatch            3.4.0.3          py36h17d0c80_0    defaults
sip                       4.19.12          py36h6538335_0    defaults
six                       1.11.0                   py36_1    defaults
smmap2                    2.0.3                     <pip>
snappy                    1.1.7                h777316e_3    defaults
snowballstemmer           1.2.1            py36h763602f_0    defaults
sortedcollections         1.0.1                    py36_0    defaults
sortedcontainers          2.0.5                    py36_0    defaults
sphinx                    1.7.9                    py36_0    defaults
sphinxcontrib             1.0                      py36_1    defaults
sphinxcontrib-websupport  1.1.0                    py36_1    defaults
sqlalchemy                1.2.11           py36hfa6e2cd_0    defaults
sqlite                    3.24.0               h7602738_0    defaults
statsmodels               0.9.0            py36h452e1ab_0    defaults
sympy                     1.2                      py36_0    defaults
tabulate                  0.8.2                      py_0    conda-forge
tblib                     1.3.2            py36h30f5020_0    defaults
tensorboard               1.10.0           py36he025d50_0    defaults
tensorflow                1.10.0          mkl_py36hb361250_0    defaults
tensorflow-base           1.10.0          mkl_py36h81393da_0    defaults
termcolor                 1.1.0                    py36_1    defaults
terminado                 0.8.1                    py36_1    defaults
testpath                  0.3.1            py36h2698cfe_0    defaults
theano                    1.0.2            py36he980bc4_0    defaults
tk                        8.6.8                hfa6e2cd_0    defaults
toolz                     0.9.0                    py36_0    defaults
tornado                   5.1              py36hfa6e2cd_0    defaults
tqdm                      4.26.0           py36h28b3542_0    defaults
traitlets                 4.3.2            py36h096827d_0    defaults
twisted                   18.7.0           py36hfa6e2cd_1    defaults
txaio                     2.9.0                      py_0    vpython
typed-ast                 1.1.0            py36hfa6e2cd_0    defaults
typing                    3.6.4                    py36_0    defaults
ujson                     1.35             py36hfa6e2cd_0    defaults
unicodecsv                0.14.1           py36h6450c06_0    defaults
urllib3                   1.23                     py36_0    defaults
vc                        14.1                 h0510ff6_3    defaults
vpnotebook                0.1.3                    py36_1    vpython
vpython                   7.4.4                    py36_0    vpython
vs2013_runtime            12.0.21005                    1    defaults
vs2015_runtime            15.5.2                        3    defaults
wcwidth                   0.1.7            py36h3d5aa90_0    defaults
webencodings              0.5.1                    py36_1    defaults
werkzeug                  0.14.1                   py36_0    defaults
wheel                     0.31.1                   py36_0    defaults
widgetsnbextension        3.4.1                    py36_0    defaults
win_inet_pton             1.0.1                    py36_1    defaults
win_unicode_console       0.5              py36hcdbd4b5_0    defaults
wincertstore              0.2              py36h7fe50ca_0    defaults
winpty                    0.4.3                         4    defaults
wrapt                     1.10.11          py36hfa6e2cd_2    defaults
wxpython                  4.0.3            py36h6538335_0    defaults
xgboost                   0.80                      <pip>
xlrd                      1.1.0                    py36_1    defaults
xlsxwriter                1.1.0                    py36_0    defaults
xlwt                      1.3.0            py36h1a4751e_0    defaults
yaml                      0.1.7                hc54c509_2    defaults
youtube-dl                2018.1.18                 <pip>
zeromq                    4.2.5                he025d50_1    defaults
zict                      0.1.3                    py36_0    defaults
zlib                      1.2.11               h8395fce_2    defaults
zope                      1.0                      py36_1    defaults
zope.interface            4.5.0            py36hfa6e2cd_0    defaults
```
</p></details>

**Comment by nehaljwani** [comment]
@thatlittleboy Try this:
```
conda config --append envs_dirs D:\user\Documents\fastai
conda activate fastai_env
```

**Comment by thatlittleboy** [comment]
@nehaljwani It works, thank you.

**Comment by axibord** [comment]
> @thatlittleboy Try this:
> 
> ```
> conda config --append envs_dirs D:\user\Documents\fastai
> conda activate fastai_env
> ```

people are awesome ! thx

**Comment by gwerbin** [comment]
Would it make sense if you could do something like this?

```python
conda create -n my-env -p /my/special/prefix
```

Using `conda config --append envs_dirs D:\user\Documents\fastai` for this purpose seems like kind of a hack, since you're saying "`D:\user\Documents\fastai` is now a standard directory where I should look for envs", which isn't exactly what is needed here. [Docs](https://docs.conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-env-directories), for reference.

**Comment by scriptonian** [comment]
Hi am having the same issue. I want to name an environment name and dont see one by the path. i did the follow

`conda create --prefix=./ml-course_env pandas numpy matplotlib scikit-learn (to create the environment)`
Then followed the instructions here.
`conda config --append envs_dirs /Users/Konstantin/Documents/code_playground/Udemy/ML-DS/ml-course/ml-course_env`

`conda activate ml-course_env`

i get (Could not find conda environment: ml-course_env)

**Comment by mforbes** [comment]
@gwerbin Yes.  That would make a lot of sense...
